### PR TITLE
feat: add quickswap integration

### PR DESCRIPTION
**Pull request by developerfred**
ref: #2645

- [x] need add provider to connect rpc network

**Network Name**
 - Matic Network
**Network URL**
https://rpc-mainnet.maticvigil.com/
**Chain ID**
 137
**Currency Symbol**
 
Screen Shot 2021-03-23 at 06 47 24
MATIC
**Block Explorer URL**
https://explorer-mainnet.maticvigil.com/

### CODE REVIEW

File: packages/maskbook/src/resources/QuickSwapIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SvgIconProps, SvgIcon } from '@material-ui/core'
+
+const svg = (
+    <svg viewBox="0 0 1000 1000">
```
Comment (septs) [inline packages/maskbook/src/resources/QuickSwapIcon.tsx:4]: plz use https://aka.pw/svgomg reduce the svg image

File: packages/maskbook/src/plugins/Trader/UI/trader/Trader.tsx [hunk 2]
```diff
@@ -329,7 +334,12 @@ export function Trader(props: TraderProps) {
                             trade={trade as TradeComputed<SwapResponse>}
                         />
                     ) : null}
-                    {[TradeProvider.UNISWAP, TradeProvider.SUSHISWAP, TradeProvider.SASHIMISWAP].includes(provider) ? (
+                    {[
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/UI/trader/Trader.tsx:337]: DRY

File: packages/maskbook/src/plugins/Trader/trader/useTradeComputed.ts [hunk 3]
```diff
@@ -25,10 +25,14 @@ export function useTradeComputed(
...
     const uniswap_ = useUniswapTrade(
         strategy,
-        [TradeProvider.UNISWAP, TradeProvider.SUSHISWAP, TradeProvider.SASHIMISWAP].includes(provider)
+        [TradeProvider.UNISWAP, TradeProvider.SUSHISWAP, TradeProvider.SASHIMISWAP, TradeProvider.QUICKSWAP].includes(
+            provider,
+        )
             ? inputAmount_
             : '0',
-        [TradeProvider.UNISWAP, TradeProvider.SUSHISWAP, TradeProvider.SASHIMISWAP].includes(provider)
+        [TradeProvider.UNISWAP, TradeProvider.SUSHISWAP, TradeProvider.SASHIMISWAP, TradeProvider.QUICKSWAP].includes(
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/Trader/trader/useTradeComputed.ts:33]: Do not repeat yourself

**Comment by Jack-Works** [comment]
hi @developerfred I mark this PR as a draft. If you feels this is ready for review please cc me thanks

**Review by Jack-Works — DISMISSED** [review]
LGTM on code style. Need a review of functionality. cc @zhouhanseng @guanbinrui

**Comment by guanbinrui** [comment]
There are CI warnings that need a fix. cc @developerfred

**Comment by septs** [comment]
image

stop, plz first fix the compile errors. thanks

**Comment by Jack-Works** [comment]
image

@septs I have to say this error looks strange. Where is it comes from?

**Review by Jack-Works — DISMISSED** [review]

File: packages/maskbook/src/extension/background-script/EthereumServices/chainState.ts [hunk 4]
```diff
@@ -104,6 +104,6 @@ export async function getUnsafeChainId(address?: string) {
 export async function getChainId(address?: string) {
     const unsafeChainId = await getUnsafeChainId(address)
     return unsafeChainId !== ChainId.Mainnet && Flags.wallet_network_strict_mode_enabled
-        ? ChainId.Mainnet
+        ? ChainId.Mainnet || ChainId.Matic
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/background-script/EthereumServices/chainState.ts:107]: Please revert this change. It will never yield to `ChainId.Matic`.

File: packages/maskbook/src/extension/background-script/EthereumServices/providers/Maskbook.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ export function createWeb3(chainId = currentMaskbookChainIdSettings.value, privK
         const weights = getConstant(CONSTANTS, 'PROVIDER_WEIGHT_LIST', chainId)
         url = urls[weights[SEED]]
     }
-    const provider = createProvider(url)
+    const provider = createProvider(url as string)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/background-script/EthereumServices/providers/Maskbook.ts:54]: Why this change?

File: packages/maskbook/src/extension/background-script/EthereumServices/chainState.ts (hunk 4 above)
Comment (developerfred) [reply packages/maskbook/src/extension/background-script/EthereumServices/chainState.ts:107]: done

File: packages/maskbook/src/extension/background-script/EthereumServices/providers/Maskbook.ts (hunk 5 above)
Comment (developerfred) [reply packages/maskbook/src/extension/background-script/EthereumServices/providers/Maskbook.ts:54]: My vscode was lintering on this type, and I am fixing it here.

**Comment by septs** [comment]
> image
> 
> @septs I have to say this error looks strange. Where is it comes from?

packages/maskbook/src/plugins/Trader/helpers/uniswap.ts

**Review by Jack-Works — DISMISSED** [review]
LTGM on code style

**Review by septs — CHANGES_REQUESTED** [review]
plz solve all errors

**Comment by Jack-Works** [comment]
@developerfred Hi I have merged with the master branch but I have no push permission to your fork repo.

**Comment by Tedko** [comment]
@developerfred hi any update? also on other bounty

**Comment by developerfred** [comment]
> @developerfred hi any update? also on other bounty

@Tedko In this I just need to resolve the type conflict between the quickswap lib, you raise the quickswap lib in npm and fix it.

**Comment by Tedko** [comment]
> > @developerfred hi any update? also on other bounty
> 
> @Tedko In this I just need to resolve the type conflict between the quickswap lib, you raise the quickswap lib in npm and fix it.

ok @developerfred how can we help on that

File: packages/maskbook/src/web3/types.ts [hunk 6]
```diff
@@ -109,8 +119,16 @@ export type TokenAssetDetailedType<
...
+// Learn more about ethereum ChainId https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+export enum ChainId {
+    Mainnet = 1,
+    Ropsten = 3,
+    Rinkeby = 4,
+    Gorli = 5,
+    Kovan = 42,
+    Matic = 137,
+    Mumbai = 80001,
+}
```
Comment (Jack-Works) [inline packages/maskbook/src/web3/types.ts:131]: Please add your new definition to `packages\shared\src\web3\index.ts` and revert this change.

File: packages/maskbook/src/web3/helpers.ts [hunk 7]
```diff
@@ -40,12 +40,15 @@ export interface Web3Constants {
...
-    chainId = ChainId.Mainnet,
+    chainId = ChainId.Mainnet || ChainId.Matic,
 ): T[K][ChainId.Mainnet] {
     return constants[key][chainId]
 }
 
-export function getAllConstants<T extends Web3Constants, K extends keyof T>(constants: T, chainId = ChainId.Mainnet) {
+export function getAllConstants<T extends Web3Constants, K extends keyof T>(
+    constants: T,
+    chainId = ChainId.Mainnet || ChainId.Matic,
```
Comment (Jack-Works) [inline packages/maskbook/src/web3/helpers.ts:50]: This expression always evaluates to `ChainId.Mainnet`.

File: packages/maskbook/src/web3/helpers.ts [hunk 8]
```diff
@@ -40,12 +40,15 @@ export interface Web3Constants {
 export function getConstant<T extends Web3Constants, K extends keyof T>(
     constants: T,
     key: K,
-    chainId = ChainId.Mainnet,
+    chainId = ChainId.Mainnet || ChainId.Matic,
```
Comment (Jack-Works) [inline packages/maskbook/src/web3/helpers.ts:43]: This expression always evaluates to `ChainId.Mainnet`.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

**Comment by Tedko** [comment]
@developerfred hi how's updates. We have integrated polygon and testing now. The Pr is out dated and need to merge from master

**Comment by Tedko** [comment]
@guanbinrui is this pr still helpful? is not lets close this pr

**Comment by guanbinrui** [comment]
Replaced by #3434.
